src/utils/string.formatter.ts

```diff
@@ -0,0 +1,17 @@
+/*
+ * String formatting functions
+ */
+const PRINTABLE_CHARS: string =
```

Comment:
Recommend using a set: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set/Set

As we can then use if PRINTABLE_CHARS.contains(char); which is more efficient

Reply:
How much more effiicent way more efficient or just little bit more?

Reply:
Assuming sets in typescript work the same as they do in Java.
You can tell if an item is in a set with a single calculation.
Whereas to tell if an item is in a list (or in this case string) it needs to do a calculation for each index of the list. (In this case 158) so by using a set we save 158 calculations for each character in the input string.

Reply:
I'm aware of this but the trouble is we are offered this set but no nice way of turning a string into a set .

Reply:
const PRINTABLE_CHARS: Set = new Set(['A', 'B', 'C']); 
This would initialise a new set with values. Then:

    for (let index = 0; index < inputStr.length; index++) {
      if (PRINTABLE_CHARS.has(inputStr.charAt(index))) {
        inputStr = inputStr.replace(inputStr.charAt(index), " ");
      }
    }

Reply:
That's what I was afraid of.

Reply:
We 'could' use https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/from
But personally I would be against this as it makes the code slightly less readable. 
Happy to get more opinions on if using this is allowed

Reply:
you could loop through the string using [...text].forEach(). see https://stackoverflow.com/a/22410812 . But i guess it would still be computationally expensive.

Reply:
Found a way to do it.

Reply:
how about this

const PRINTABLE_CHARS: string = "etc...."
const mySet: Set<string> = new Set<string>([...PRINTABLE_CHARS]);

Reply:
Already sorted this - I don't know whetehr there isn't some operation going on under the hood for this - I think the set has been chosen because we want to start with a set.